lightningd/peer_control.c

```diff
@@ -1754,6 +1755,59 @@ static const struct json_command listpeers_command = {
 /* Comment added to satisfice AUTODATA */
 AUTODATA(json_command, &listpeers_command);
 
+static void json_add_scb(struct lightningd *ld,
+			  struct json_stream *response,
+			  struct channel *c)
+{
+	u8 *msg = towire_static_chan_backup(tmpctx, 
+										c->scb->id, 
+										&c->scb->cid, 
+										&c->scb->node_id, 
+										&c->scb->addr, 
+										&c->scb->funding);
+
+	json_add_hex_talarr(response, NULL,
+			msg);
+}
+
+/*This will return a SCB for all the channels currently loaded 
+	in the in-memory channel*/
+static struct command_result *json_hex_scb(struct command *cmd,
+					     const char *buffer,
+					     const jsmntok_t *obj UNNEEDED,
+					     const jsmntok_t *params)
+{
+	struct json_stream *response;
+
+	if (!param(cmd, buffer, params, NULL))
+        return command_param_failed();
+	response = json_stream_success(cmd);
+
+	json_array_start(response, "scb");
+
+	/* FIXME: Filter out some channels based on their state? */
+
+	struct peer *peer;
+	struct channel *channel;
+	list_for_each(&cmd->ld->peers, peer, list)
+		list_for_each(&peer->channels, channel, list)
+			json_add_scb(cmd->ld, response, channel);
+
+	json_array_end(response);
+
+	return command_success(cmd, response);
+}
+
+static const struct json_command hex_scb_command = {
+	"hex-scb",
+	"backup",
+	json_hex_scb,
+	"Returns SCB of all the channels currently present in the DB"
+};
```

Comment:
Yes, I am completely removing this RPC and adding it in `listpeers`...

Reply:
Or maybe keep this just to make it easy for fetching scb by the plugin? Because we don't need SCB for uncommitted and unsaved channels and `listpeers` have them as well, Extracting it in RPC_scan is just too much string-parsing :sweat_smile:

Reply:
we need to think also about the performance of these command`listpeers`, maybe can be improved by #5022 but I don't know yet!